crates/miden-tx/src/executor/exec_host.rs

```diff
@@ -109,18 +110,65 @@ where
     ) -> Result<(), TransactionKernelError> {
         let pub_key = process.get_stack_word(0);
         let msg = process.get_stack_word(1);
+
         let signature_key = Hasher::merge(&[pub_key, msg]);
 
         let signature = if let Ok(signature) =
             process.advice_provider().get_mapped_values(&signature_key)
         {
             signature.to_vec()
         } else {
-            let signing_inputs = SigningInputs::Blind(msg);
+            // Retrieve transaction summary commitments from the advice provider.
+            // The commitments are stored as a contiguous array of field elements with the following
+            // layout:
+            // - commitments[0..4]:  SALT
+            // - commitments[4..8]:  OUTPUT_NOTES_COMMITMENT
+            // - commitments[8..12]: INPUT_NOTES_COMMITMENT
+            // - commitments[12..16]: ACCOUNT_DELTA_COMMITMENT
+            let commitments = process
+                .advice_provider()
+                .get_mapped_values(&msg)
+                .map_err(|err| TransactionKernelError::TransactionSummaryError(Box::new(err)))?;
+
+            if commitments.len() != 16 {
+                return Err(TransactionKernelError::TransactionSummaryError(
+                    "Expected 4 words for transaction summary commitments".into(),
+                ));
+            }
+
+            let extract_word = |start: usize| -> Word {
+                Word::from([
+                    commitments[start],
+                    commitments[start + 1],
+                    commitments[start + 2],
+                    commitments[start + 3],
+                ])
+            };
+
+            let salt = extract_word(0);
+            let output_notes_commitment = extract_word(4);
+            let input_notes_commitment = extract_word(8);
+            let account_delta_commitment = extract_word(12);
+            let tx_summary = build_tx_summary(
+                self.base_host(),
+                salt,
+                output_notes_commitment,
+                input_notes_commitment,
+                account_delta_commitment,
+            )
+            .map_err(|err| TransactionKernelError::TransactionSummaryError(Box::new(err)))?;
+
+            debug_assert_eq!(
+                msg,
+                tx_summary.to_commitment(),
+                "transaction summary doesn't commit to the expected message"
+            );
```

Comment:
Should we turn this into an error, e.g. `TransactionKernelError::TransactionSummaryError`? I think that would be slightly better than a debug assertion.